library/func_misc.py

```diff
@@ -153,3 +153,55 @@ def Call(self, pos_args, named_args):
 
         raise error.TypeErr(x, 'len() expected Str, List, or Dict',
                             loc.Missing)
+
+
+class Join(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        li = r.PosList()
+
+        delim = r.NamedStr('delim', '')
```

Comment:
Hm not sure about this named arg?

- Do Python or JS have it?
- If we have it, we should document it (doc/ref is kinda ready), and have test coverage

I would be inclined to leave it out, it seems like `join(x, '+')`  is relatively clear

Reply:
Ah yeah. I'm OK to drop this. IIRC nothing depends on it right now, so it's just here because it was part of the original impl (https://github.com/oilshell/oil/pull/1712/files#diff-1b57cb0469c5f87781d4b8ac574803c6ca501351c865e4d8a10f04343d33700aL38)